### ISSUE TITLE
Usage of scripting in `@ExpectedDataSet`

### ISSUE THREAD

**Issue by coiouhkc** [database-rider/database-rider#358]
Hi @rmpestano ,
is it possible to use smth like `groovy: $value < 10` in `@ExpectedDataSet` (yml), where `$value` is the real field value?

**Comment by rmpestano** [comment]
Hi @coiouhkc,  you can use [`regex` in expected dataset](https://github.com/database-rider/database-rider#371-expected-dataset-with-regular-expressions) to match ranges, ex:

```
USER:
  id: "regex: ([0-9])"
```

You can also use word boundaries to [limit the range](https://stackoverflow.com/questions/7861689/checking-a-number-range-with-regular-expressions):

```
id: "regex: \b(\d|1[0-2])\b"
```

The above regex would match against 0-12, explanation:
```
\b(         //The beginning of a word (or number), followed by either
   \d       //   Any digit 0 through 9
|           //Or
   1[0-2]   //   A 1 followed by any digit between 0 and 2.
)\b         //The end of a word
```

I hope it helps

**Comment by coiouhkc** [comment]
Hi @rmpestano, i'm using regex in the project right now, but i think scripting would be a simpler (and also more maintainable) and more flexible option. 
If you agree and if it's possible to add this feature in dbrider directly (as compared to adding it into dbunit), I could work on a PR.

**Comment by rmpestano** [comment]
Do you have more examples how it would work?

For example, if we use [scriptable datasets](https://github.com/database-rider/database-rider#scriptable-datasets) like in `groovy: $value < 10` what happens is that we evaluate that when we get the table value, see [here](https://github.com/database-rider/database-rider/blob/8b840367c8ddcbb8c7a3d2f5c23dc788192d19f9/rider-core/src/main/java/com/github/database/rider/core/api/dataset/ScriptableTable.java#L53) and [here](https://github.com/database-rider/database-rider/blob/8b840367c8ddcbb8c7a3d2f5c23dc788192d19f9/rider-core/src/main/java/com/github/database/rider/core/assertion/DataSetAssert.java#L66)  and in your case it would evaluate to an expression. 

Unless if you can identify it's an expression and then try to evaluate it using the actual value here 
https://github.com/database-rider/database-rider/blob/8b840367c8ddcbb8c7a3d2f5c23dc788192d19f9/rider-core/src/main/java/com/github/database/rider/core/assertion/DataSetAssert.java#L66-L67
 
But I'm not sure

**Comment by coiouhkc** [comment]
Another example where this functionality could come in handy is where one wants to make assertions for generated values based on certain invariants, e.g.
* dates ($value > now() - 1.days && $value < now() + 1.years -1.days)
* numbers ($value.isEven())
* jsonb ( JSON.equals($value, '{"some":"json", "maybeOrdered": true}'


Regarding code you mentioned - exactly, that's what I was talking about
* check for 'groovy:' or 'js:'
* add `$value` for actual value interpolation
* call the ScriptEngine to check the assertion

**Comment by rmpestano** [comment]
That's interesting and might work! waiting for your PR and let me know if you need any help on that.

Thank you!

**Comment by rmpestano** [comment]
This feature is available in v`1.31.0`, thank you @coiouhkc!
